src/content/packages-and-plugins/swift-package-manager/for-plugin-authors.md

```diff
@@ -0,0 +1,155 @@
+---
+title: Swift Package Manager for plugin authors
+description: How to add Swift Package Manager compatibility to iOS and macOS plugins
+diff2html: true
+---
+
+:::warning
+Flutter is migrating to [Swift Package Manager][] to manage iOS and macOS native
+dependencies.
+Flutter's support of Swift Package Manager is under development.
+The implementation might change in the future.
+Swift Package Manager support is only available on the [`main` channel][].
+Flutter continues to support CocoaPods.
+:::
+
+Flutter's Swift Package Manager integration has several benefits:
+
+1. **Access to the Swift package ecosystem**.
+   Flutter plugins can use the growing ecosystem of [Swift packages][]! 
+2. **Simplifies Flutter installation**.
+   Swift Package Manager is bundled with Xcode.
+   In the future, you won’t need to install Ruby and CocoaPods to target iOS or
+   macOS.
+
+If you find a bug in Flutter's Swift Package Manager support,
+[open an issue][].
+
+[Swift Package Manager]: https://www.swift.org/documentation/package-manager/
+[`main` channel]: /release/upgrade#switching-flutter-channels
+[Swift packages]: https://swiftpackageindex.com/
+[open an issue]: {{site.github}}/flutter/flutter/issues/new?template=2_bug.yml
+
+{% include docs/swift-package-manager/how-to-enable-disable.md %}
+
+## How to add Swift Package Manager support to an existing Flutter plugin
+
+This guide shows how to add Swift Package Manager support to a plugin that
+already supports CocoaPods.
+This ensures the plugin is usable by all Flutter projects.
+
+Flutter plugins should _both_ Swift Package Manager and CocoaPods until further
+notice.
+
+Swift Package Manager adoption will be gradual.
+Plugins that don't support CocoaPods won't be usable by projects that haven't
+migrated to Swift Package Manager yet.
+Plugins that don't support Swift Package Manager can cause problems for projects
+that have migrated.
```

Comment:
It might be nice to explicitly say here that we expect to remove support for CocoaPods at some point in the future, so adding SPM support will future-proof plugins. That will make it more clear why a plugin author should do this work.

Reply:
Agreed.

@atsansone What would be the best way to state this while following the timeless [GSG](https://developers.google.com/style/timeless-documentation)?

> Flutter expects to remove support for CocoaPods. Adding Swift Package Manager support future-proofs your plugins.

I think this would be a valuable sentence to add, but it goes against the GSG guidance.